### ISSUE TITLE
Support tracing in ExecutionEngine

### ISSUE THREAD

**Issue by jackm321** [pytorch/glow#2927]
It would be helpful debugging performance issues with backends if the ExeuctionEngine would support tracing so that one doesn't have to manually create a HostManager. Maybe if ExecutionEngine and HostManager are eventually merged this will be a non-issue but it would be great to have support for this.

**Comment by nickgg** [comment]
@jackm321 What level of tracing do you mean here? Right now it looks like you can enable tracing pretty easily by setting a TraceContext on the provided ExecutionContext - but since the ExecutionEngine doesn't use the full runtime you'll only get DeviceManager down events.

**Comment by jackm321** [comment]
Yeah we were using this last week to get trace events from the DeviceManager from unit tests. This was useful functionality but required modifying each test by setting TraceContext and then dumping the traces in the test body. It would be nice if this could be enabled in ExecutionEngine itself. I'm not sure if it will be broadly useful or not though so not high pri.

**Comment by nickgg** [comment]
Seems pretty easy to add a flag like the -glow-dump-debug-trace one in Onnxifi for the unit tests, but might make sense to wait until the EE is refactored to use HostManager - did that get in yet?
